tmt/convert.py

```diff
@@ -201,23 +218,27 @@ def read_datafile(path, filename, datafile, types, testinfo=None):
         testinfo = datafile
     # Beaker task name
     try:
-        beaker_task = re.search(regex_task, testinfo).group(1)
+        search_result = re.search(regex_task, testinfo)
+        assert search_result is not None
```

Comment:
Are we absolutely sure `search_result` is going to be a valid value? What if `filename` is `Makefile` but said Makefile is missing `Name: ...` line because the user removed it? We'd drop a nasty assert traceback on the user's head when we could instead raise a nice `GeneralError` saying "sorry, could find line matching this pattern, are you sure it's there?"

Reply:
Thank you. I removed all asserts and added conditional instead. However, I removed a couple of `try except` constructions as they were not needed anymore